scripts/main.js

```diff
@@ -1,31 +1,75 @@
-const createLog = (event) => {
+import DeleteButton from "./deleteWorkout.js";
+
+const fields = [
+  {
+    id: "time-input",
+    name: "time",
+  },
+  {
+    id: "workout-input",
+    name: "workout",
+  },
+  {
+    id: "date-input",
+    name: "date",
+  },
+  {
+    id: "action",
+    name: "action",
+  },
+];
+
+let totalTime = 0;
+
+const getWorkoutData = () => {
+  const data = {};
+  fields.forEach((field) => {
+    const element = document.querySelector(`#${field.id}`);
+    if(element !== null){
+   
+      data[field.name] = element.value;
+    }
+  });
+
+  return data;
+};
+
+function initializeWorktouTable() {
+  const workoutTable = document.querySelector("#table");
+  let tableHeaders = "";
+
+  fields.forEach((field) => {
+    const tableHeader = `<th>${field.name}</th>`;
+    tableHeaders += tableHeader;
+  });
```

Comment:
Você pode utilizar o [reduce](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) para economizar umas linhas de código :D, a ideia dele é reduzir o valor de um array a um outro tipo (string, outro array, objeto e etc).

Nesse caso estamos transformando o array fields em uma string. O primeiro parametro é a função que vai fazer a conversão e o segundo é o valor inicial.

```suggestion
  const tableHeaders = fields.reduce((tableHeaders, field) => `${tableHeaders}<th>${field.name}</th>` , "")
```